### PR TITLE
Task/apps 3162 user identification notification

### PR DESCRIPTION
**Pull request by anzioka**
PR:
- Display notification when device is unlocked. On tapping notification, user is shown a screen where they can select "Target Child" or "Other" as the current device user. The selected value is saved as a shared preference. The previously selected value is also saved on the device. If user swipes away notification, subsequent usage events are not assigned to any specific user (**Perhaps we should modify this to behavior to instead assign a default user)**

- Preferences screen with an option to turn on/off on-device-unlock notifications
- Usage data is tagged with current device user. 

Screenshot_20211017-202847

Screenshot_20211017-201425

Screenshot_20211017-201431

Screenshot_20211017-201446

Screenshot_20211017-201453

Screen Shot 2021-10-17 at 8 40 41 PM

### CODE REVIEW

**Comment by solomontang** [comment]
Have the researchers been asked what kind of behavior they want if the notification is dismissed? I can imagine it gets dismissed a lot for users who like to press "clear all" in their notifications

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -45,4 +45,27 @@
     <string name="settings">Settings</string>
     <string name="uploading_data">Uploading data</string>
 
+    <!-- Phone wake up notification   -->
+    <string name="on_wake_notification_title">User Identification</string>
+    <string name="on_wake_notification_message">Tap on notification to identify device user</string>
```
Comment (solomontang) [inline app/src/main/res/values/strings.xml:50]: Can just say "Tap to identify device user"

Comment (anzioka) [reply app/src/main/res/values/strings.xml:50]: much better, thanks!

**Comment by anzioka** [comment]
> Have the researchers been asked what kind of behavior they want if the notification is dismissed? I can imagine it gets dismissed a lot for users who like to press "clear all" in their notifications

We are in the process of reaching out to confirm expected behavior. For now I am assigning the ol.user property type an empty string whenever the notification is dismissed. Will wait for confirmation and make changes if necessary.

File: app/src/main/java/com/openlattice/chronicle/receivers/lifecycle/UnlockPhoneReceiver.kt [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+            val notificationBuilder = NotificationCompat.Builder(context, CHANNEL_ID)
+                .setSmallIcon(R.drawable.ic_stat_notification)
+                .setColor(ContextCompat.getColor(context, R.color.colorPrimary))
+                .setCustomContentView(layout)
+                .setCustomBigContentView(layout)
+                .setPriority(NotificationCompat.PRIORITY_HIGH)
+                .setContentIntent(pendingIntent)
+                .setDeleteIntent(createOnDismissedIntent())
+                .setDefaults(NotificationCompat.DEFAULT_VIBRATE)
```
[the remark this answers is not in the document]
Comment (anzioka) [reply app/src/main/java/com/openlattice/chronicle/receivers/lifecycle/UnlockPhoneReceiver.kt:69]: Researchers have not been specific on this. Yes we should probably put this in settings, but would prefer to keep it as is for now.

**Comment by anzioka** [comment]
Researcher confirmed that we can leave user blank when notification is swiped away

**Review by solomontang — DISMISSED** [review]

**Review by UnsungHero97 — APPROVED** [review]

**Review by solomontang — APPROVED** [review]

**Review by giuliacampana — APPROVED** [review]

**Pull request merged**